src/main/scala/org/broadinstitute/dsde/workbench/sam/google/DisableUserMessageReceiver.scala

```diff
@@ -0,0 +1,41 @@
+package org.broadinstitute.dsde.workbench.sam.google
+
+import cats.effect.unsafe.implicits.global
+import com.google.cloud.pubsub.v1.{AckReplyConsumer, MessageReceiver}
+import com.google.pubsub.v1.PubsubMessage
+import com.typesafe.scalalogging.LazyLogging
+import org.broadinstitute.dsde.workbench.model._
+import org.broadinstitute.dsde.workbench.sam.model.UserStatus
+import org.broadinstitute.dsde.workbench.sam.service.UserService
+import org.broadinstitute.dsde.workbench.sam.util.{OpenCensusIOUtils, SamRequestContext}
+
+/** Created by srubenst on 02/04/21.
+  */
+class DisableUserMessageReceiver(userService: UserService) extends MessageReceiver with LazyLogging {
+  override def receiveMessage(message: PubsubMessage, consumer: AckReplyConsumer): Unit =
+    OpenCensusIOUtils
+      .traceIO("DisableUsersMonitor-PubSubMessage", SamRequestContext()) { samRequestContext =>
+        val userId = WorkbenchUserId(message.getData.toStringUtf8)
+        logger.info(s"received disable user message: $userId")
+        userService
+          .disableUser(userId, samRequestContext)
+          .redeem(handleFailure(_, consumer), handleDisableUserResponse(userId, _, consumer))
+      }
+      .unsafeRunSync()
```

Comment:
This unsafeRunSync is a little smelly, just want to make sure we want to block receiving these messages. I could see a situation where we get backed up and get dead letters.

Reply:
It is intended. This subscriber stuff is running within a Java framework and is expected to be blocking. Managing the back pressure is part of the google library.